test/diagnostics_tests.jl

```diff
@@ -0,0 +1,136 @@
+#= 
+    Unit tests for ClimaCoupler Diagnostics module
+=#
+using Test
+using Dates
+using ClimaCore: InputOutput
+using ClimaComms
+using ClimaCoupler: Utilities
+using ClimaCoupler.TimeManager: EveryTimestep, Monthly
+using ClimaCoupler.TestHelper: create_space
+import ClimaCoupler.Diagnostics:
+    get_var,
+    init_diagnostics,
+    accumulate_diagnostics!,
+    save_diagnostics,
+    TimeMean,
+    DiagnosticsGroup,
+    pre_save,
+    post_save,
+    save_time_format
+
+
+FT = Float64
```

Comment:
could add loop `for FT in (Float32, Float64)` over all testsets to test both FTs

Reply:
Yeah, I thought about it too, but then saw that it's not really testing any fundamental dependencies on the FT, so I figured Float64 suffices.  Trying to cut down on unnecessary CI ;)